### DISCUSSION TITLE
Should we enforce uniqueness on form slugs?

### DISCUSSION THREAD

**Discussion by DavidBiddle** [govuk-forms/forms#175] [Ideas]
Is this something we should do? We've talked about it before. Making this thread to track any further discussion we have on this.

Pros:
- helps identify duplicate forms
- prevents confusion where a user has two forms with the same name
- means that we could potentially use them as friendly URLs
- means that we could use them instead of IDs in other places where uniqueness is required and readability is an advantage (e.g. the form-specific feature flags https://github.com/alphagov/forms-runner/pull/865)
- has advantages for reporting
- makes it easier for users to spot when they're on the wrong form

Cons:
- if there are valid use cases for duplicate form names, they won't be possible
- we'll need to do some design around how we tell users about this restriction
- we'll need to do something about any existing duplicate form names
- GOV.UK pages don't enforce uniqueness
- would potentially get tricky if/when we open up to local government (since they all provide similar services)

Questions:
- will this actually encourage users to write unique form names, or will they just call their form `{duplicate form name} 2`/`{duplicate form name} (new)`/etc? 

Alternatives:
- a report for flagging duplicates and manually contacting the owners to sort them out
- hints/warnings which tell the user there's another form with the same slug, but don't stop them making the form live

**Comment by lfdebrux** [comment]
It would be good to get input from the consultancy team and content designers on this.

The [GOV.UK URL standards](https://www.gov.uk/guidance/content-design/url-standards-for-gov-uk) definitely seem to suggest using the form slug is better than our current opaque identifiers, but I wonder if maybe it's easier with their content design teams being more centralised? It would be interesting to know though if they have considered this issue before.

It is worth noting though that for a published form the URL shouldn't usually be shared, instead the start page of the service should be, so what the URL is shouldn't affect form fillers as much as the content of the form itself.

(Idea that just occurred while writing the above paragraph, what if we could make the form slug the same as the GOV.UK start page slug 🤔).
